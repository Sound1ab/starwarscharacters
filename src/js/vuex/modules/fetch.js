import axios from 'axios';
import {fetchMachine} from '@/js/vuex/FSM/fetch-machine';
import {transition} from '@/js/vuex/fsm-transition';
import {starWars} from '@/js/vuex/api';

export const FETCH_TRANSITION = 'FETCH_TRANSITION';
export const CREATE_CANCEL_TOKEN = 'CREATE_CANCEL_TOKEN';
export const CANCEL_OUTGOING_REQUEST = 'CANCEL_OUTGOING_REQUEST';
export const FETCH_DATA = 'FETCH_DATA';

function initialState () {
	return {
		state: fetchMachine.initialState,
		cancelToken: null,
		data: {}
	};
}

const state = initialState();

const actions = {
	[FETCH_TRANSITION]: transition.bind(null, fetchMachine),
	[CREATE_CANCEL_TOKEN] ({commit, dispatch}, {params}) {
		let CancelToken = axios.CancelToken;
		let source = CancelToken.source();
		commit(CREATE_CANCEL_TOKEN, source);
		dispatch(FETCH_TRANSITION, {type: 'TOKEN_CREATED', params});
	},
	[CANCEL_OUTGOING_REQUEST] ({state}) {
		if (state.cancelToken) {
			state.cancelToken.cancel();
		}
	},
	[FETCH_DATA] ({commit, dispatch}, {params: {type, page = 0, query = ''}}) {
		const queryString = type === 'SEARCH' && query.length > 0
			? `?page=${page}`
			: `?search=${query}`;
		axios.get(`${starWars['PEOPLE']}${queryString}`)
			.then(res => {
				if (res.status !== 200) {
					throw new Error(res.statusText);
				}
				commit(FETCH_DATA, res.data.results);
				dispatch(FETCH_TRANSITION, {type: 'SUCCESS'});
			})
			.catch(() => {
				dispatch(FETCH_TRANSITION, {type: 'FAILURE'});
			});
	}
};

const mutations = {
	[FETCH_TRANSITION] (state, nextState) {
		state.state = nextState;
	},
	[CREATE_CANCEL_TOKEN] (state, cancelToken) {
		state.cancelToken = cancelToken;
	},
	[FETCH_DATA] (state, payload) {
		state.data = payload;
	}
};

export default {
	state,
	mutations,
	actions
};
