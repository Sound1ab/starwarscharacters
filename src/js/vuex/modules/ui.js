export const SHOW_LOADING = 'SHOW_LOADING';
export const HIDE_LOADING = 'HIDE_LOADING';
export const SHOW_CONFIRMATION = 'SHOW_CONFIRMATION';

function initialState () {
	return {
		loading: false,
		confirmation: {
			state: false,
			value: '',
			message: ''
		}
	};
}

const state = initialState();

const actions = {
	[SHOW_LOADING] ({commit}) {
		commit(SHOW_LOADING, true);
	},
	[HIDE_LOADING] ({commit}) {
		commit(HIDE_LOADING, false);
	},
	[SHOW_CONFIRMATION] ({commit}, payload) {
		commit(SHOW_CONFIRMATION, payload);
		setTimeout(() => {
			commit(SHOW_CONFIRMATION, initialState().confirmation);
		}, 1000);
	}
};

const mutations = {
	[SHOW_LOADING] (state, payload) {
		state.loading = payload;
	},
	[HIDE_LOADING] (state, payload) {
		state.loading = payload;
	},
	[SHOW_CONFIRMATION] (state, payload) {
		state.loading = payload;
	}
};

export default {
	state,
	mutations,
	actions
};
