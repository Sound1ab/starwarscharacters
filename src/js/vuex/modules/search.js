import {searchMachine} from '@/js/vuex/FSM/search-machine';
import {transition} from '@/js/vuex/fsm-transition';
import {FETCH_TRANSITION} from '@/js/vuex/modules/fetch';

export const SEARCH_TRANSITION = 'SEARCH_TRANSITION';
export const START_TIMER = 'START_TIMER';
export const UPDATE_SEARCH = 'UPDATE_SEARCH';
export const DISPATCHING_SEARCH = 'DISPATCHING_SEARCH';

let timeout;

function initialState () {
	return {
		state: searchMachine.initial,
		query: ''
	};
}

const state = initialState();

const actions = {
	[SEARCH_TRANSITION]: transition.bind(null, searchMachine),
	[START_TIMER] ({dispatch}, {params}) {
		clearTimeout(timeout);
		timeout = setTimeout(() => {
			dispatch(SEARCH_TRANSITION, {
				type: 'TIMER_COUNTDOWN_PASSED',
				params
			});
		}, 500);
	},
	[UPDATE_SEARCH] ({commit}, {params: {query = ''}}) {
		commit(UPDATE_SEARCH, query);
	},
	[DISPATCHING_SEARCH] ({dispatch}, {params}) {
		dispatch(FETCH_TRANSITION, {
			type: 'FETCH_DATA',
			params
		});
		dispatch(SEARCH_TRANSITION, {type: 'SEARCH_DISPATCHED'});
	}
};

const mutations = {
	[SEARCH_TRANSITION] (state, nextState) {
		state.state = nextState;
	},
	[UPDATE_SEARCH] (state, query) {
		state.query = query;
	}
};

export default {
	state,
	mutations,
	actions
};
