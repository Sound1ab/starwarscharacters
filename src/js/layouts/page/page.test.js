/* eslint-disable */
import {shallow, createLocalVue} from '@vue/test-utils';
import {createRenderer} from 'vue-server-renderer';
import Vuex from 'vuex';
import Page from './page';
import {FETCH_TRANSITION} from '../../vuex/modules/fetch';

const localVue = createLocalVue();
localVue.use(Vuex);

let mountComponent = (store) => shallow(Page, {
	localVue,
	store,
	mocks: {
		$route: {
			params: {
				id: 2
			}
		}
	}
});

describe('Pagination', () => {
	let wrapper;
	let store;
	let actions;
	beforeEach(() => {
		actions = {
			[FETCH_TRANSITION]: jest.fn()
		};
		store = new Vuex.Store({
			actions,
			getters: {
				PARSED_API_QUERY: () => {},
				CURRENT_PAGE: () => 2
			},
			state: {
				fetch: {
					data: ''
				}
			}
		});
		wrapper = mountComponent(store);
		jest.clearAllMocks();
	});
	afterEach(() => {
		wrapper.destroy();
	});
	describe('htmlStructure', () => {
		it('html snapshot matches', () => {
			expect(wrapper.html()).toMatchSnapshot();
		});
	});
	describe('component', () => {
		it('vm - is instantiated', () => {
			expect(wrapper.isVueInstance()).toBe(true);
		});
	});
});
