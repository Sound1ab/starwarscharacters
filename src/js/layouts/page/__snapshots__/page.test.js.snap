// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination htmlStructure html snapshot matches 1`] = `<div class="page"></div>`;
